zcash_client_backend/src/data_api/wallet.rs

```diff
@@ -1146,29 +1236,30 @@ where
                 SentTransactionOutput::from_parts(output_index, recipient, value, memo)
             });
 
-    let transparent_outputs =
-        transparent_output_meta
-            .into_iter()
-            .map(|(recipient, addr, value)| {
-                let script = addr.script();
-                let output_index = build_result
-                    .transaction()
-                    .transparent_bundle()
-                    .and_then(|b| {
-                        b.vout
-                            .iter()
-                            .enumerate()
-                            .find(|(_, tx_out)| tx_out.script_pubkey == script)
-                    })
-                    .map(|(index, _)| index)
-                    .expect(
-                        "An output should exist in the transaction for each transparent payment.",
-                    );
+    let txid: [u8; 32] = build_result.transaction().txid().into();
+    assert_eq!(
+        transparent_output_meta.len(),
+        build_result
+            .transaction()
+            .transparent_bundle()
+            .map_or(0, |b| b.vout.len()),
+    );
 
-                SentTransactionOutput::from_parts(output_index, recipient, value, None)
-            });
+    #[allow(unused_variables)]
+    let transparent_outputs = transparent_output_meta.into_iter().enumerate().map(
```

Comment:
The `enumerate()` here is making an assumption that the transaction builder is not shuffling the order of transparent outputs. The old code instead looked up the output index matching the recipient (instead assuming that the transaction contained no more than one transparent output per transparent recipient).

Document this assumption here so that we know to change it if necessary.

Reply:
I don't think there will ever be a reason to shuffle the transparent outputs. But I will document it.